charts/ExploreView.tsx

```diff
@@ -27,19 +49,95 @@ const DUMMY_JSON_CONFIG = {
 //
 // -@jasoncrawford 2 Dec 2019
 
-export class ExploreView extends React.Component<{ bounds: Bounds }> {
+interface ExploreProps {
+    bounds: Bounds
+}
+
+@observer
+export class ExploreView extends React.Component<ExploreProps> {
     static bootstrap({ containerNode }: { containerNode: HTMLElement }) {
         const rect = containerNode.getBoundingClientRect()
         const bounds = Bounds.fromRect(rect)
         return ReactDOM.render(<ExploreView bounds={bounds} />, containerNode)
     }
 
+    // This is different from the chart's concept of chart type because it includes "WorldMap" as
+    // an option, and doesn't include certain chart types we don't support right now, such as
+    // scatter plots
+    @observable chartType: string = ChartType.LineChart
+
+    chart: ChartConfig
+
+    constructor(props: ExploreProps) {
+        super(props)
+
+        const chartProps = new ChartConfigProps()
+        extend(chartProps, DUMMY_JSON_CONFIG)
+        this.chart = new ChartConfig(chartProps)
+
+        // We need these updates in an autorun because the chart config objects aren't really meant
+        // to be recreated all the time. They aren't pure value objects and have behaviors on
+        // instantiation that include fetching data over the network. Instead, we rely on their
+        // observable properties, and on this autorun block to connect them to the Explore controls.
+        // -@jasoncrawford 2019-12-04
+        autorun(() => {
+            this.chart.tab = this.tab
+            this.chart.props.type = this.configChartType
+            this.chart.props.hasMapTab = this.isMap
+            this.chart.props.hasChartTab = !this.isMap
+        })
```

Comment:
It feels a bit icky having these synced up like this. This ensures changes to ExploreView get applied to the ChartView, but not the other way around... but I think it's fine for what we are doing. 

One potential issue is that the tab will get switched whenever any of the properties change, which is again probably fine for now, but just something to watch out for, since the user can switch to the source, data & download tabs.

On the nitpicky side, `autorun` returns a disposer that should run in `componentWillUnmount` like here: https://github.com/owid/owid-grapher/blob/b3b70fbb8eb7fad2d02935d416effe562a88e133/admin/client/VariableSelector.tsx#L382-L384

We always store it in an attribute: https://github.com/owid/owid-grapher/blob/b3b70fbb8eb7fad2d02935d416effe562a88e133/admin/client/VariableSelector.tsx#L364

Reply:
> This ensures changes to ExploreView get applied to the ChartView, but not the other way around

What do you mean? What's an example?

> the tab will get switched whenever any of the properties change

Not sure what this refers to either?

> autorun returns a disposer that should run in componentWillUnmount

Great feedback, thanks for catching that! Will fix

Reply:
> > This ensures changes to ExploreView get applied to the ChartView, but not the other way around
> 
> What do you mean? What's an example?

I meant `this.configChartType` is not guaranteed to be `== this.chart.props.type`, and same for `this.tab`. They shouldn't be used to check the current state basically, they are only pushed to the chart config when any property used within the `autorun` changes. The user could've selected the sources tab for example – `config.props.tab` would be `"sources"` while the value of `this.tab` would be `"chart"`.

One example I didn't think of earlier and that complicates things a bit is that when a user selects a Line Chart and collapses the timeline to a single year, the chart becomes a `DiscreteBar`. This is not reflected in `chart.props.type` right now, but the information needs to reach the explorer UI somehow so that the right type is highlighted.

> > the tab will get switched whenever any of the properties change
> 
> Not sure what this refers to either?

`autorun` runs whenever any of the properties used within it change. So a change to any of `this.tab`, `this.isMap` or `this.chartConfigType` will force a re-sync for all properties, and as I mentioned above, they may not necessarily be in sync.

---

As I said, this is all fine for now! Just expressing what I'm thinking of when I look at the code.

Reply:
Hmm, good point, I'd forgotten about the sources/data tabs. I experimented with this. The result of this is just that if you click Sources or Data, and then you change the chart type, it removes the overlay and pops you back to the chart/map. I think that's fine behavior.